src/validation/tests/recursive_validation_tests.rs

```diff
@@ -884,3 +884,107 @@ mod inheritance {
         ");
     }
 }
+
+mod type_aliases {
```

Comment:
i would like to see tests where the chain is broken by adding a reference. Both in validation and that it codegens maybe even lit tests